src/components/common/animations/SpinnerCheckMark/index.js

```diff
@@ -0,0 +1,28 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import SpinnerCheckMark from './SpinnerCheckMark'
+
+export default props => {
+  const { loading, success, onFinish } = props
+  const [isStarting, setIsStarting] = useState(false)
+  const [isSuccess, setIsSuccess] = useState(false)
+  const onFinishHandle = useCallback(() => {
+    onFinish && onFinish()
+    setIsStarting(false)
+    setIsSuccess(false)
+    onFinish && onFinish()
+  })
+
+  useEffect(() => {
```

Comment:
why do we need useEffect if success + loading are comming from props? if props change it will rerender the object, so not sure why we need useEffect also.

Reply:
because when loading is finished we need to continue our animation and smooth finish it. that's why I've added useEffect for loading and success and set values to state

Reply:
isSuccess - because we need to set success to false (without changing on parent component) when onFinishHandle called.

isStarting - because when loading is finished we need to continue our animation and smooth finish it. that's why I've added useEffect for loading and success and set values to state

Reply:
@AlexeyKosinski how about my last commit, does it work?

Reply:
no, it is not working. we need to use hooks
![1](https://user-images.githubusercontent.com/36693645/72903750-7d6c4f80-3d36-11ea-8592-f898a813ffed.gif)